highlighters/main/main-highlighter.zsh

```diff
@@ -195,7 +195,13 @@ _zsh_highlight_main__stack_pop() {
 _zsh_highlight_highlighter_main_paint()
 {
   # Before we even 'emulate -L', save the user's options
-  local -A useroptions; useroptions=("${(@kv)options}")
+  local -A useroptions
+  if zmodload -e zsh/parameter; then
+    useroptions=("${(@kv)options}")
+  else
+    useroptions=(${(s: :)${(f)"$(setopt -m \*; setopt)"}/%on/off} ${(s: :)${(f)"$(setopt)"}/%/ on})
```

Comment:
Setting _all_ options!?  I can't imagine how that might behave unexpectedly...

(For starters, when I do `setopt -m \*` interactively it dumps vcs_info lines before the next prompt, since it sets XTRACE.)